sql/parse/parse.go

```diff
@@ -2423,6 +2430,147 @@ func havingToHaving(ctx *sql.Context, having *sqlparser.Where, node sql.Node) (s
 	return plan.NewHaving(cond, node), nil
 }
 
+// windowToWindow will apply a WINDOW clause to a plan.Window node,
+// replacing named window references with their window definitions.
+// Performs semantic validation of window compatibility/validity.
+// TODO: separate plan building from replacement and semantic validation
+func windowToWindow(ctx *sql.Context, window sqlparser.Window, node sql.Node) (sql.Node, error) {
+	w, ok := node.(*plan.Window)
```

Comment:
Fair to say that the top level node passed here must be a plan.Window? If so, change the signature to enforce that